packages/react-storage/src/components/StorageBrowser/views/LocationsView/LocationsView.tsx

```diff
@@ -57,54 +50,27 @@ const LocationsEmptyMessage = () => {
 
 export function LocationsView({
   className,
-  onNavigate,
+  onNavigate: onNavigateProp,
 }: LocationsViewProps): React.JSX.Element {
-  const [{ data, isLoading, hasError }, handleList] = useLocationsData();
-
-  const { result, nextToken } = data;
-  const resultCount = result.length;
-  const hasNextToken = !!nextToken;
-
-  // initial load
-  React.useEffect(() => {
-    handleList({
-      options: { ...DEFAULT_LIST_OPTIONS, refresh: true },
-    });
-  }, [handleList]);
-
-  const onPaginateNext = () =>
-    handleList({
-      options: { ...DEFAULT_LIST_OPTIONS, nextToken },
-    });
-
   const {
-    currentPage,
-    handlePaginateNext,
-    handlePaginatePrevious,
-    handleReset,
-  } = usePaginate({ onPaginateNext, pageSize: DEFAULT_PAGE_SIZE });
-
-  const { disableNext, disablePrevious, disableRefresh, range } =
-    listViewHelpers({
-      currentPage,
-      hasNextToken,
-      isLoading,
-      pageSize: DEFAULT_PAGE_SIZE,
-      resultCount,
-      hasError,
-    });
+    pageItems,
+    hasError,
+    isPaginatePreviousDisabled,
+    isPaginateNextDisabled,
+    page,
+    isLoading,
+    onRefresh,
+    onPaginateNext,
+    onPaginatePrevious,
+    onNavigate,
+  } = useLocationsView({ onNavigate: onNavigateProp });
 
   // FIXME: Eventually comes from useView hook
```

Comment:
this can be removed now?

Reply:
I think we'll have some churn here as we finalize Composables and the ControlsContext interface. It's better if we handle this in a follow up